File: Leetcode/1110 Delete Nodes And Return Forest.ts
function delNodes(root: TreeNode | null, toDelete: number[]): Array<TreeNode | null> {
    const toDeleteSet = new Set<number>(toDelete);
    const forest: (TreeNode | null) [] = [];
    function dfs(root: TreeNode | null) {
        if (root === null) {
            return null;
        }
        root.left = dfs(root.left); 
        root.right = dfs(root.right);
        if (!toDeleteSet.has(root.val)) {
            return root;
        }
        if (root.left !== null) {
            forest.push(root.left);
        }
        if (root.right !== null) {
            forest.push(root.right);
        }
        return null;
    }
    const newRoot = dfs(root);
    if (newRoot !== null) {
        forest.push(newRoot);
    }
    return forest;
}

File: Leetcode/1110 Delete Nodes And Return Forest.py
class Solution:
    def delNodes(self, root, to_delete):
        to_delete = set(to_delete)
        forest = []
        def dfs(root):
            if root is None:
                return None
            root.left = dfs(root.left) 
            root.right = dfs(root.right)
            if root.val not in to_delete:
                return root
            if root.left:
                forest.append(root.left)
            if root.right:
                forest.append(root.right)
            return None
        new_root = dfs(root)
        if new_root is not None:
            forest.append(new_root)
        return forest
